docs/design/2018-08-10-restore-dropped-table.md

```diff
@@ -0,0 +1,49 @@
+
+## Rationale
+
+Let's take a look at the workflow of the `DROP TABLE` statement. The `DROP TABLE` statement first removes the dropping table meta data from the coresponding database meta data. After the schemas are synced by all the TiDB instances, in `worker.deleteRange`, TiDB will insert a deleted range of the first row key to the end row key of the dropping table into the table `mysql.gc_delete_range`. At most `max(gcDefaultRunInterval, gcLifeTimeKey)` time later, the GC worker will delete the table data finally. 
+
+The meta data of the table is not really deleted. The meta key format is `Table:table_id`. As long as we can find out the ID of the dropped table, we can recover the table information. The `admin show ddl jobs` statement can retrieve the table ID:
+
+```
++-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
+| JOBS                                                                                                                                                                                                                          | STATE     |
++-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
+| ID:44, Type:drop table, State:synced, SchemaState:none, SchemaID:1, TableID:39, RowCount:0, ArgLen:0, start time: 2018-08-11 11:23:53.308 +0800 CST, Err:<nil>, ErrCount:0, SnapshotVersion:0                                 | synced    |
++-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
+```
+
+As you can see, if we can restore the table before the GC worker deletes the table data, we can restore the table completely. If the table data is deleted, we can only restore an empty table.
+
+Before we run `ADMIN RESTORE TABLE table_id`, you must:
+* Ensure that no GC task is running and then you can figure this by using TiDB logs and metrics.
+* Increase `tikv_gc_life_time` to a sufficient value.
+
+## Compatibility
+
+It's a new command and will not lead to compatibility issues.
+
+## Implementation
+
+1. `ADMIN RESTORE TABLE table_id` will enqueue a new DDL job to TiDB general DDL queue.
```

Comment:
Could we put it to the beginning of the queue?

Reply:
Job in the general queue will be handled quickly, maybe treat it as a normal job is ok.